notebooks/09_Monitoring/src/07_01_Iris_Pipeline_Monitoring/model.py

```diff
@@ -0,0 +1,216 @@
+    roc_file = os.path.join('/', 'roc.csv')
+    with file_io.FileIO(roc_file, 'w') as f:
+        df_roc.to_csv(f, columns=['fpr', 'tpr', 'thresholds'], header=False, index=False)
+
+    return cm_file, roc_file, vocab
+
+
+def tf_training(args, X_train, y_train, X_test, y_test):
+    model = tf.keras.Sequential([
+        tf.keras.layers.Dense(10, activation='relu', input_shape=(4,)),  # input shape required
+        tf.keras.layers.Dense(32, activation='relu'),
+        tf.keras.layers.Dense(64, activation='relu'),
+        tf.keras.layers.Dense(128, activation='relu'),
+        tf.keras.layers.Dense(2)
+    ])
+
+    model.compile(optimizer=args.tf_nn_optimizer,
+                  loss=args.tf_nn_loss_func,
+                  metrics=[args.tf_nn_metrics])
+
+    time_hash = datetime.datetime.now().strftime("%Y%m%d-%H%M%S")
+    log_dir = "/logs/fit/" + time_hash
+    tensorboard_callback = tf.keras.callbacks.TensorBoard(log_dir=log_dir)
+
+    model.fit(x=X_train,
+              y=y_train,
+              epochs=args.tf_nn_epoch,
+              validation_data=(X_test, y_test),
+              callbacks=[tensorboard_callback])
+
+    return model, log_dir, time_hash
+
+
+def upload_s3(args, cm_file, roc_file ,log_dir, time_hash):
+    AWS_REGION = args.aws_region
+    s3_client = boto3.client('s3', region_name=AWS_REGION)
+
+    try:
+        # upload cm file to S3
+        cm_file_name = cm_file
+        # bucket = 'eks-kubeflow-workshop-data'
+        bucket = args.s3_bucket
+        cm_object_name = 'iris-example/confusion_matrix.csv'
+        s3_cm_file = 's3://' + bucket + '/' + cm_object_name
+        s3_client.upload_file(cm_file_name, bucket, cm_object_name)
+
+        roc_object_name = 'iris-example/roc.csv'
+        s3_roc_file = 's3://' + bucket + '/' + roc_object_name
+        s3_client.upload_file(roc_file, bucket, roc_object_name)
+
+        # upload tb log dir to S3
+        tb_object_name = 'iris-example/tb-logs'
+        s3_tb_file = 's3://' + bucket + '/' + tb_object_name
+        # s3_tb_file = 's3://eks-kubeflow-workshop-data/iris-example/tb-logs'
+        for path, subdirs, files in os.walk(log_dir):
+            path = path.replace("\\", "/")
+            directory_name = path.replace(log_dir, "")
+            for file in files:
+                s3_client.upload_file(os.path.join(path, file), bucket,
+                                      'iris-example/tb-logs/' + time_hash + directory_name + '/' + file)
+    except ClientError as e:
+        logging.info("ERROR IN S3 UPLOADING!!!!!!")
+        logging.ERROR(e)
+
+    logging.info("S3 bucket path is: s3://{}/iris-example".format(bucket))
+
+    return s3_cm_file, s3_roc_file, s3_tb_file
+
+
+def write_metadata(args, s3_cm_file, s3_roc_file, s3_tb_file, vocab):
+    metadata = {
+        'outputs': [
+            {
+                'source': 's3://{}/iris-example/README.md'.format(args.s3_bucket),
```

Comment:
What does the README come come?

Reply:
You pass full S3 uri  for s3_cm_file, s3_roc_file, you'd better make this one consistent with others as well.

Reply:
Yes, I forgot to upload it to S3, which should be done.

Reply:
I'm considering not use `README.md` but create a `Markdown` file with metrics specified and upload it on S3, I think this might makes more sense.

Reply:
I created a `Markdown` file and upload it on S3 similar to `cm_file`, `tb_file`, etc. And then markdown artifact can read from S3 bucket.